core/device/fdbased/open_unix.go

```diff
@@ -5,14 +5,19 @@ package fdbased
 import (
 	"fmt"
 	"os"
+	"runtime"
 
 	"github.com/xjasonlyu/tun2socks/v2/core/device"
 	"github.com/xjasonlyu/tun2socks/v2/core/device/iobased"
 )
 
 func open(fd int, mtu uint32, offset int) (device.Device, error) {
 	f := &FD{fd: fd, mtu: mtu}
-
+	// fd offset in ios
+	// https://stackoverflow.com/questions/69260852/ios-network-extension-packet-parsing/69487795#69487795
+	if offset == 0 && runtime.GOOS == "ios" {
```

Comment:
Would it make more sense to put this if condition here:?
https://github.com/xjasonlyu/tun2socks/blob/a821cc483cd18648a882f7f0a88aa73b48beb20f/engine/parse.go#L70

Reply:
seems better now.